src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -1,31 +1,65 @@
 package com.es.phoneshop.web;
 
-import com.es.phoneshop.model.product.ArrayListProductDao;
-import com.es.phoneshop.model.product.Product;
-import com.es.phoneshop.model.product.ProductDao;
-import com.es.phoneshop.model.product.ProductNotFoundException;
+import com.es.phoneshop.cart.Cart;
+import com.es.phoneshop.model.product.*;
 
 import javax.servlet.ServletConfig;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
 
 
 public class ProductDetailsPageServlet extends HttpServlet {
     private ProductDao productDao;
+    private CartService cartService;
 
     @Override
     public void init(ServletConfig config) throws ServletException {
         super.init(config);
         productDao = ArrayListProductDao.getInstance();
+        cartService = DefaultCartService.getInstance();
     }
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        String productId = request.getPathInfo();
-        request.setAttribute("product", productDao.getProduct(Long.valueOf(productId.substring(1))));
+        Long productId = parseProductId(request);
+        request.setAttribute("product", productDao.getProduct(productId));
+        request.setAttribute("cart", cartService.getCart(request));
         request.getRequestDispatcher("/WEB-INF/pages/product.jsp").forward(request, response);
     }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String quantityString = request.getParameter("quantity");
+        Long productId = parseProductId(request);
+        int quantity;
+        try {
+            NumberFormat format = NumberFormat.getInstance(request.getLocale());
+            quantity = format.parse(quantityString).intValue();
+        } catch (ParseException ex) {
+            request.setAttribute("error", "Not a Number");
+            doGet(request, response);
+            return;
+        }
+        Cart cart = cartService.getCart(request);
+        try {
```

Comment:
I'd combine this into single try with multiple catch, just better approach. But it's also fine, no changes needed